group/views.py

```diff
@@ -40,15 +40,20 @@
 
 def get_group(group_id):
     try:
-        group = Group.objects.get(id = group_id)
+        group = Group.objects.get(id=group_id)
     except Group.DoesNotExist:
-        logger.warning('Group: Can not edit group %s! Group is not exist!' % group_id)
-        return render(
-            request,
-            'index/404.html',
-            {'error_message': 'Group: Can not edit group %s! Group is not exist!' % group_id})
+        logger.warning('Group: Can not edit group %s! Group does not exist!' % group_id)
+        raise Http404('Group: Can not edit group %s! Group does not exist!' % group_id)
     return group
 
+def get_announce(announce_id):
+    try:
+        announce = Announce.objects.get(id=announce_id)
+    except Announce.DoesNotExist:
+        logger.warning('Announce: Can not edit announce %s! Announce does not exist!' % announce_id)
+        raise Http404('Announce: Can not edit announce %s! Announce does not exist!' % announce_id)
+    return announce
+
 def get_running_contest(request, group_id):
```

Comment:
I suggest to use django build-in filter (instead of for loop).
```running_contests = group.trace_contest.filter(start_time__lte=now, end_time__gte=now)```